ta4j-core/src/main/java/org/ta4j/core/BacktestingResult.java

```diff
@@ -0,0 +1,93 @@
+        this.totalProfitLossPercentage = new ProfitLossPercentageCriterion().calculate(series, tradingRecord);
+        this.totalProfit = new TotalProfit2Criterion().calculate(series, tradingRecord);
+        this.totalLoss = new TotalLossCriterion().calculate(series, tradingRecord);
+        this.profitTradeCount = new NumberOfWinningTradesCriterion().calculate(series, tradingRecord);
+        this.lossTradeCount = new NumberOfLosingTradesCriterion().calculate(series, tradingRecord);
+        this.breakEvenTradeCount = new NumberOfBreakEvenTradesCriterion().calculate(series, tradingRecord);
+    }
+
+    public String getStrategyName() {
+        return strategy.getName();
+    }
+
+    public Strategy getStrategy() {
+        return strategy;
+    }
+
+    public TradingRecord getTradingRecord() {
+        return tradingRecord;
+    }
+
+    public Num getTotalProfitLoss() {
+        return totalProfitLoss;
+    }
+
+    public Num getTotalProfitLossPercentage() {
+        return totalProfitLossPercentage;
+    }
+
+    /**
+     * Print to console the backtesting result
+     *
+     * @param printTrades - Print trades from the backtesting result
+     * @param series      - The time series
+     */
+    public void printBacktestingResult(boolean printTrades, TimeSeries series) {
+        System.out.println("------------ " + getStrategyName() + " ------------");
+        System.out.println("Total trades: " + tradingRecord.getTradeCount());
+        System.out.println("Total profit: " + totalProfit + " Trade count: " + profitTradeCount);
+        System.out.println("Total loss: " + totalLoss + " Trade count: " + lossTradeCount);
+        System.out.println("Break event trade count: " + breakEvenTradeCount);
+        System.out.println("Total profitLoss: " + totalProfitLoss);
+        System.out.println("Total profitLoss: " + round(totalProfitLossPercentage, 1) + "%");
+        if (printTrades) {
+            tradingRecord.getTrades().stream().forEach(trade -> printTrade(trade, series));
+            Order lastOrder = tradingRecord.getLastOrder();
+            if (lastOrder.getType() == Order.OrderType.BUY) {
+                Bar lastOrderBar = series.getBar(lastOrder.getIndex());
+                System.out.println("Trade entry: " + lastOrderBar.getSimpleDateName() + ", exit: ?.");
+            }
+        }
+        System.out.println();
+    }
+
+    private void printTrade(Trade trade, TimeSeries series) {
+        Bar entryBar = series.getBar(trade.getEntry().getIndex());
+        Bar exitBar = series.getBar(trade.getExit().getIndex());
+        System.out.println("Trade entry: " + entryBar.getSimpleDateName() + ", exit: " + exitBar.getSimpleDateName() + ". Profit: " + trade.getProfit());
+    }
+
+    @Override
+    public int compareTo(BacktestingResult that) {
+        return this.totalProfitLossPercentage.compareTo(that.getTotalProfitLossPercentage());
+    }
+
+    private static double round(Num value, int precision) {
```

Comment:
I am not sure if it good place for this kind of util method

Reply:
I agree. Any suggestion to where to place it?